homeassistant/components/wallbox/number.py

```diff
@@ -26,15 +28,64 @@
 from .entity import WallboxEntity
 
 
+def min_charging_current_value(coordinator: WallboxCoordinator) -> float:
+    """Return the minimum available value for charging current."""
+    if (
+        coordinator.data[CHARGER_DATA_KEY][CHARGER_PART_NUMBER_KEY][0:2]
+        in BIDIRECTIONAL_MODEL_PREFIXES
+    ):
+        return cast(float, (coordinator.data[CHARGER_MAX_AVAILABLE_POWER_KEY] * -1))
+    return 0
+
+
+async def async_set_charging_current_value(
+    coordinator: WallboxCoordinator, value: float
+) -> None:
+    """Set the value for charging current."""
+    await coordinator.async_set_charging_current(value)
+
+
+async def async_set_energy_price_value(
+    coordinator: WallboxCoordinator, value: float
+) -> None:
+    """Set the value for energy price."""
+    await coordinator.async_set_energy_cost(value)
+
+
 @dataclass
-class WallboxNumberEntityDescription(NumberEntityDescription):
+class WallboxNumberEntityDescriptionMixin:
+    """Load entities from different handlers."""
+
+    max_value_fn: Callable[[WallboxCoordinator], float]
+    min_value_fn: Callable[[WallboxCoordinator], float]
+    set_value_fn: Callable[[WallboxCoordinator, float], Awaitable[None]]
+
+
+@dataclass
+class WallboxNumberEntityDescription(
+    NumberEntityDescription, WallboxNumberEntityDescriptionMixin
+):
     """Describes Wallbox number entity."""
 
 
 NUMBER_TYPES: dict[str, WallboxNumberEntityDescription] = {
     CHARGER_MAX_CHARGING_CURRENT_KEY: WallboxNumberEntityDescription(
         key=CHARGER_MAX_CHARGING_CURRENT_KEY,
         translation_key="maximum_charging_current",
+        max_value_fn=lambda coordinator: cast(
+            float, coordinator.data[CHARGER_MAX_AVAILABLE_POWER_KEY]
+        ),
+        min_value_fn=min_charging_current_value,
+        set_value_fn=async_set_charging_current_value,
```

Comment:
```suggestion
        set_value_fn=lambda coordinator: coordinator.async_set_charging_current,
```

Reply:
nice solution, didn't know this was an option